lib/Service/AssistantService.php

```diff
@@ -132,6 +133,28 @@ private function sanitizeInputs(string $type, array $inputs): array {
 					if (count($inputs) !== 2) {
 						throw new \Exception('Invalid input(s)');
 					}
+					break;
+				}
+			case 'OCA\\ContextChat\\TextProcessing\\ContextChatTaskType':
+				{
+					if ((count($inputs) !== 1 && count($inputs) !== 3)
+						|| !isset($inputs['prompt'])
+						|| !is_string($inputs['prompt'])
+					) {
+						throw new \Exception('Invalid input(s)');
+					}
+
+					if (count($inputs) === 3) {
+						if (!isset($inputs['scopeType']) || !is_string($inputs['scopeType'])
+							|| !isset($inputs['scopeList']) || !is_array($inputs['scopeList'])) {
+							throw new \Exception('Invalid input(s)');
+						}
+						$inputs['scopeList'] = array_filter(
+							array_map(fn (array $scope) => $scope['id'] ?? null, $inputs['scopeList']),
+							fn ($scope) => is_string($scope),
+						);
```

Comment:
Will the scope list be loaded correctly when displaying an existing task?
Wouldn't it be simpler to keep the structure of `$inputs['scopeList']` intact so it is straightforward to load it in `<ContextChatInputForm>`?

Reply:
hmm, it did work in the Assistant modal but noticed just now that it does not when pulled from the result notification. Will de-couple the UI and actual data shortly.